core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/JsonCodec.scala

```diff
@@ -275,6 +286,18 @@ object JsonCodec {
     RuntimeImageType.stringToRuntimeImageType.get(s).toRight(s"invalid RuntimeImageType ${s}")
   )
 
+  implicit val cloudContextDecoder: Decoder[CloudContext] = Decoder.instance { x =>
+    for {
+      cloudProvider <- x.downField("cloudProvider").as[CloudProvider]
+      context <- cloudProvider match {
+        case CloudProvider.Gcp =>
+          x.downField("cloudResource").as[GoogleProject].map(p => CloudContext.Gcp(p))
+        case CloudProvider.Azure =>
+          x.downField("cloudResource").as[ManagedResourceGroup].map(p => CloudContext.Azure(p))
```

Comment:
Are we still thinking of embedding `"tenantId,subscriptionId,managedResourceGroupName"` as the Azure cloud context? I'm wondering If the decoder should split that string into separate fields..

Reply:
ah yea...forgot this needs to be refactored for that